src/language-js/printer-estree.js

```diff
@@ -3003,12 +3003,14 @@ function printPathNoParens(path, options, print, args) {
       // Keep comma if the file extension is .tsx and
       // has one type parameter that isn't extend with any types.
       // Because, otherwise formatted result will be invalid as tsx.
+      const grandParent = path.getNode(2);
       if (
         parent.params &&
         parent.params.length === 1 &&
         options.filepath &&
         options.filepath.match(/\.tsx/) &&
```

Comment:
While you're at it, please improve this regexp a bit: `/\.tsx$/i`
Also it's better to rewrite the last two lines using `test`: `/\.tsx$/i.test(options.filepath)`

Reply:
Fix from b961cdc .